(1524) Force sync stats broadcast on startup so the metrics view has initial values

garvankeeley opened the pull request:
When the ClientStumblerService is initialized, force a single read of the sync stats, and broadcast this for the metrics view to update on startup.
Note that in the app the UI components are initialized before the service.

https://github.com/mozilla/MozStumbler/issues/1524

garvankeeley committed:
When the ClientStumblerService is initialized, force a single read of
the sync stats, and broadcast this for the metrics view to update on
startup.
Note that in the app the UI components are initialized before the service.

crankycoder commented:
Minor nit I just noticed that `PersistedStats::incrementSyncStats` has a call to System.currentTimeMillis, that should be changed to ISystemClock.

crankycoder commented:
review is done.

crankycoder commented:
Hmm. Hold up actually.  I just whipped up a small testcase and exposed a minor bug.

crankycoder committed:
Added a test and forced a 'zero' record to be emitted if no sync stats
are available so that we can force reset any metrics.  This is
important for things like remoteviews and desktop widgets.

crankycoder commented:
@garvankeeley I sent a PR to your branch.

garvankeeley commented:
looking at it now

Commit:
Merge pull request #7 from crankycoder/garvan/stats

Added a test and forced a 'zero' record to be emitted if no sync stats

garvankeeley commented:
Thanks for the test case @crankycoder! Much gooder now.
